### DISCUSSION TITLE
Using Hetzner Load Balancer to pin sessions to a server

### DISCUSSION THREAD

**Discussion by TeemuKoivisto** [mysticaltech/terraform-hcloud-kube-hetzner#955] [Q&A]
Hey!

Wonderful project, makes Kubernetes seem almost doable.

I have a question about the built-in Hetzner load balancers - are they objectively better than rolling out your own LBs? I want to enforce sticky sessions by adding a header from CloudFlare worker to pin the request to specific server but it doesn't seem it's possible with the built-in LBs.

I am considering adding my own Traefik LB to the mix but I hope it can handle as many active sessions and throughput as the built-in.

**Comment by TeemuKoivisto** [comment]
Oh this is possible in Hetzner but just didn't show up in Google results directly. My bad!

**Reply by mysticaltech** [reply]
Yep, but not sure this is how ingress on kubernetes should be used. However, glad you found what you were looking for!

**Reply by TeemuKoivisto** [reply]
Oh do people setup their own DNS with k8s? That seems scary 😅 . And actually, I think I will have to make my own custom LB solution since I really need granular control over the sessions. Otherwise, CloudFlare workers & DNS is working fine as the routing layer to my app. 

I can get geo-based redirects to the correct servers (without exposing IPs) and customize it as much as I want. Pricing of course is a question mark but so far it seems pretty cheap.

**Reply by mysticaltech** [reply]
I see, for such granular comtrol, using cilium as a cni would be great, along with any of the imgress controller, traefik or nginx. But if you are satisfied with what you built, it's fine! Like the old adage goes, "if it ain't broken, don't try to fix it"! 😂
